### PR TITLE
Refactor user add feature in groups edit page

### PR DESCRIPTION
**Pull request by DonOmalVindula**
### Purpose

- This PR will refactor the user addition feature in the groups section to better utilize BE APIs than performing actions on the front-end. This also includes some UI enhancements to the user listing table.

### Related Issues

- N/A

### Related PRs

- N/A

### Checklist

- [x] e2e cypress tests locally verified. (for internal contributers)
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any) - N/A
- [ ] Relevant backend changes deployed and verified - N/A
- [ ] [Unit tests](/docs/testing/UNIT_TESTING.md) provided. (Add links if there are any) - N/A
- [ ] [Integration tests](https://github.com/wso2/product-is/tree/master/modules/integration) provided. (Add links if there are any) - N/A

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines
- [ ] Ran FindSecurityBugs plugin and verified report - N/A
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

### CODE REVIEW

File: features/admin.groups.v1/components/edit-group/add-group-user-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+interface AddGroupUserModalProps extends IdentifiableComponentInterface {
+    group: GroupsInterface;
+    handleAddUserSubmit: (selectedUsers: UserBasicInterface[]) => void;
+    handleCloseAddNewGroupModal: () => void;
+    showAddNewUserModal: boolean;
+    userstore: string;
+}
+
+export const AddGroupUserModal: FunctionComponent<AddGroupUserModalProps> = (
+    props: AddGroupUserModalProps): ReactElement => {
```
Comment (pavinduLakshan) [inline features/admin.groups.v1/components/edit-group/add-group-user-modal.tsx:56]:
```suggestion
export const AddGroupUserModal: FunctionComponent<AddGroupUserModalProps> = (
    props: AddGroupUserModalProps
): ReactElement => {
```

File: features/admin.groups.v1/components/edit-group/add-group-user-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+     * Handles the user list fetch request error.
+     */
+    useEffect(() => {
+        if (!userListFetchRequestError) {
+            return;
+        }
+
+        if (userListFetchRequestError.response
+            && userListFetchRequestError.response.data
+            && userListFetchRequestError.response.data.description) {
```
Comment (pavinduLakshan) [inline features/admin.groups.v1/components/edit-group/add-group-user-modal.tsx:131]:
```suggestion
        if (userListFetchRequestError?.response?.data?.description) {
```

File: modules/i18n/src/translations/en-US/portals/groups.ts [hunk 3]
```diff
@@ -57,6 +57,22 @@ export const groups: groupsNS = {
                 }
             },
             subHeading: "View assigned roles for the group."
+        },
+        users: {
+            addUsersModal: {
+                heading: "Update Group Users",
```
Comment (pavinduLakshan) [inline modules/i18n/src/translations/en-US/portals/groups.ts:63]: How about something like `Manage Group Members`?

File: features/admin.groups.v1/components/edit-group/add-group-user-modal.tsx (hunk 2 above)
Comment (DonOmalVindula) [reply features/admin.groups.v1/components/edit-group/add-group-user-modal.tsx]: Fixed with b365305b21a319b26ebb47986884e422cf9819c6

File: modules/i18n/src/translations/en-US/portals/groups.ts (hunk 3 above)
Comment (DonOmalVindula) [reply modules/i18n/src/translations/en-US/portals/groups.ts]: Fixed with b365305b21a319b26ebb47986884e422cf9819c6

File: features/admin.users.v1/utils/user-management-utils.ts [hunk 4]
```diff
@@ -134,6 +134,40 @@ export class UserManagementUtils {
             .some((schemaItem: ProfileSchemaInterface) =>
                 schemaItem.name === UserManagementConstants.SCIM2_SCHEMA_DICTIONARY.get("DISPLAY_NAME"));
     }
+
+    /* Resolves username.
+    *
+    * @returns Username for the user avatar.
+    */
```
Comment (pavinduLakshan) [inline features/admin.users.v1/utils/user-management-utils.ts:141]:
```suggestion
   /* Resolves username.
    *
    * @param user - user details
    *
    * @returns Username for the user avatar.
    */
```

**Review by pavinduLakshan — DISMISSED** [review]

File: features/admin.users.v1/utils/user-management-utils.ts (hunk 4 above)
Comment (DonOmalVindula) [reply features/admin.users.v1/utils/user-management-utils.ts]: Fixed with 3e4e4b867a5f958a2ae1d2a161149323c67d8099

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**